### ISSUE TITLE
Development setup for Ubuntu, Mac, and Windows

### ISSUE THREAD

**Issue by resinneublem** [pyfa-org/Pyfa#912]
Just sharing my notes from setting up Pyfa on all environments. I would really prefer to figure out a way to do this without installing the dependencies globally. VirtualEnv looks interesting, but I'm a bit of a python noob. I also remember had trouble with one or two dependencies last time I tried but I didn't dig too deep into it.

Posting this in case it might add to https://github.com/pyfa-org/Pyfa/wiki/Setting-Up-Development-Environment or even be pushed into the Readme itself.

Ubuntu
------------

```
sudo apt-get update
sudo apt-get install python-wxgtk3.0 python-wxtools python-matplotlib python-requests python-dateutil python-sqlalchemy
```

Mac
----

1. Install homebrew http://brew.sh/
2. Install python 2.7 from homebrew: `brew install python`
2. `brew install wxpython`
3. `pip install sqlalchemy dateutils requests matplotlib`

Windows
------------
Basically followed https://github.com/pyfa-org/Pyfa/wiki/Setting-Up-Development-Environment except I followed the directions on matplotlib to install it too.

1. https://www.python.org/ftp/python/2.7.11/python-2.7.11.msi
2. https://sourceforge.net/projects/wxpython/files/wxPython/3.0.2.0/wxPython3.0-win32-3.0.2.0-py27.exe/download?use_mirror=jaist
3. Install dependencies

```
C:\Python27\python.exe -m pip install --upgrade pip
C:\Python27\python.exe -m pip install sqlalchemy
C:\Python27\python.exe -m pip install dateutils
C:\Python27\python.exe -m pip install requests

:: from http://matplotlib.org/users/installing.html
C:\Python27\python -m pip install -U pip setuptools
C:\Python27\python -m pip install matplotlib
```

**Comment by blitzmann** [comment]
A couple days ago I was actually looking at Vagrant for unified dev setup.

**Comment by Ebag333** [comment]
pyCharm will handle most of the dependencies once we get the requirements.txt all merged in. The big one it won't is wx. 😠

For Windows its fairly easy, just run the installation. Mac and Linux seem to be the painful ones there.

**Comment by blitzmann** [comment]
Are you guys able to make wiki pages? Or request PRs for wiki pages? Not entirely sure how github handles that stuff, but this is good info to have in a wiki

**Comment by blitzmann** [comment]
Closing this as it's now a lot easier to get things set up thanks for wxPhoenix having binary wheels for Windows, Mac, and popular Linux distros.
